sale_order_currency_rate/views/res_company.xml

```diff
@@ -0,0 +1,12 @@
+<odoo>
+    <record id="view_res_company_form_inherit" model="ir.ui.view">
+        <field name="name">res.company.form.inherit</field>
+        <field name="model">res.company</field>
+        <field name="inherit_id" ref="base.view_company_form" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='company_registry']" position="after">
+                <field name="sale_show_currency_rate" />
```

Comment:
You don't need to add the field in res.company view.

You need to add it to res.config.settings view in sales configuration.